src/data.rs

```diff
@@ -287,6 +304,7 @@ mod printing {
                 Visibility::Restricted(ref path) => {
                     tokens.append("pub");
                     tokens.append("(");
+                    tokens.append("in");
```

Comment:
Please add a special case when printing that omits "in" if the path is self or super. This is what Rust does. You can see it by running cargo expand on the following code.

```rust
pub(self) fn f() {}
pub(in self) fn g() {}

fn main() {}
```